### DISCUSSION TITLE
Unable to find image

### DISCUSSION THREAD

**Discussion by etmartinkazoo** [basecamp/kamal#688] [Q&A] [answered]
I have been using Kamal with Rails apps and love it.

Recently one of my apps failed to build "container unhealthy"

I ran `kamal healthcheck` and got the following:
```
ERROR (SSHKit::Command::Failed): Exception while executing on host <redacted IP Address>: docker exit status: 125
docker stdout: Nothing written
docker stderr: Unable to find image 'user/image:hash' locally
docker: Error response from daemon: manifest for user/image:hash not found: manifest unknown: manifest unknown.
See 'docker run --help'.
```

I have been searching for a solution for days and would love some guidance if anyone has had a similar issue.

Thank you!

**Comment by nickhammond** [comment]
@etmartinkazoo Is the `user/image:hash` portion what it's actually saying or redacted? This is for a role and not an accessory?

**Reply by etmartinkazoo** [reply]
The user/image:hash is redacted.

Here is an example: `myusername/coolcodeproject`

It is really strange

**Reply by nickhammond** [reply]
@etmartinkazoo If you run a `docker image ls` on the server I'm guessing it's not in that list? Any other hashes?

**Comment by etmartinkazoo** [comment]
No, it seems to be a local docker issue. My other Kamal projects are working so I am truly perplexed.

**Reply by nickhammond** [reply]
Do you have more from the output? You can try resetting your local builder with `kamal build remove` but it's not really clear what the full run looks like to tell if that's relevant.

**Reply by etmartinkazoo** [reply]
I really appreciate you helping. Here is some more output:
```
ERROR bin/rails aborted!
ActiveSupport::MessageEncryptor::InvalidMessage:
/rails/config/environment.rb:5:in `<main>'
Tasks: TOP => db:prepare => db:load_config => environment
(See full trace by running task with --trace)
 ERROR {
  "Status": "unhealthy",
  "FailingStreak": 2,
  "Log": [
    {
      "Start": "2024-02-21T22:28:00.06343526Z",
      "End": "2024-02-21T22:28:00.287730272Z",
      "ExitCode": 1,
      "Output": "  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current\n                                 Dload  Upload   Total   Spent    Left  Speed\n\r  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0\r  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0\ncurl: (7) Failed to connect to localhost port 3000 after 7 ms: Couldn't connect to server\n"
    },
    {
      "Start": "2024-02-21T22:28:01.28976489Z",
      "End": "2024-02-21T22:28:01.463927714Z",
      "ExitCode": 1,
      "Output": "  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current\n                                 Dload  Upload   Total   Spent    Left  Speed\n\r  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0\r  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0\ncurl: (7) Failed to connect to localhost port 3000 after 7 ms: Couldn't connect to server\n"
    }
  ]
}
```

**Reply by etmartinkazoo** [reply]
Here is some more:
```
ERROR (SSHKit::Command::Failed): Exception while executing on host <IP Address>: docker exit status: 125
docker stdout: Nothing written
docker stderr: Unable to find image '<user>/<image>:812d99ab118162281d753221871ec84d0a34e839_uncommitted_3605a0e352a18206' locally
docker: Error response from daemon: manifest for <user>/<image>:812d99ab118162281d753221871ec84d0a34e839_uncommitted_3605a0e352a18206 not found: manifest unknown: manifest unknown.
See 'docker run --help'.
```

**Reply by nickhammond** [reply]
Oh, so the healthcheck is probably going to fail when you run it one-off like that. Notice the `812d99ab118162281d753221871ec84d0a34e839_uncommitted_3605a0e352a18206` on the image name. Have you deployed with that specific version? You can see the image versions on your server by running `kamal app images`.

The main issue looks to be rails credentials decryption which is the error that's raising in your entrypoint script `ActiveSupport::MessageEncryptor::InvalidMessage:`. Are you using Rails credentials? How can you tell your master key is getting loaded properly?

**Reply by etmartinkazoo** [reply]
It is strange because I have the correct mater.key in the config directory. I have that image on the server but cannot boot from it...

**Reply by nickhammond** [reply, marked as answer]
You have config/master.key and config/credentials.yml.enc in your running container? I'd try removing the `db:prepare` line and just do a `kamal app exec -i "/bin/bash"` and poke around in there to confirm the credentials are in there and the correct ones.

**Reply by etmartinkazoo** [reply]
@nickhammond - Thank you so much. Your help was extremely helpful in getting to the bottom of this. It ended up being an error in the application and your last suggestions helped in that debugging. Thank you so very much!

**Reply by nickhammond** [reply]
@etmartinkazoo No problem, glad you got it figured out! Can you mark this one answered?
